### PR TITLE
Deprecate futures ticker

### LINKED ISSUE #6622: chore(gossipsub): deprecate futures-ticker
## Description

from rust-libp2p, to address [RUSTSEC-2024-0384 ](https://rustsec.org/advisories/RUSTSEC-2024-0384.html). Use `futures-timer` and `Delay` instead. I'll post a PR soon

### PR DESCRIPTION
**Pull request by hopinheimer**
fixes #6622

### CODE REVIEW

**Review by jxs — APPROVED** [review]
Thanks! Can you address the Clippy issues?

File: beacon_node/lighthouse_network/gossipsub/src/behaviour.rs [hunk 1]
```diff
@@ -29,8 +29,8 @@ use std::{
     time::Duration,
 };
 
-use futures::StreamExt;
-use futures_ticker::Ticker;
+#[allow(unused_imports)]
```
Comment (jxs) [inline beacon_node/lighthouse_network/gossipsub/src/behaviour.rs]: huh! Why do we need this?

File: beacon_node/lighthouse_network/gossipsub/src/behaviour.rs [hunk 2]
```diff
@@ -3427,14 +3424,15 @@ where
         }
 
         // update scores
-        if let Some((peer_score, _, interval)) = &mut self.peer_score {
-            while let Poll::Ready(Some(_)) = interval.poll_next_unpin(cx) {
+        if let Some((peer_score, _, delay)) = &mut self.peer_score {
+            if delay.poll_unpin(cx).is_ready() {
                 peer_score.refresh_scores();
```
Comment (jxs) [inline beacon_node/lighthouse_network/gossipsub/src/behaviour.rs:3429]: we need to reset the `Delay` here as well

**Review by michaelsproul — APPROVED** [review]
Looks good to me too

**Comment by michaelsproul** [comment]
@mergify queue

**Pull request merged**